CRM/Core/Config.php

```diff
@@ -377,8 +377,14 @@ public static function clearTempTables($timeInterval = FALSE): void {
     }
     if (!empty($tables)) {
       $table = implode(',', $tables);
-      // drop leftover temporary tables
-      CRM_Core_DAO::executeQuery("DROP TABLE $table");
+      // If a User Job references the table do not drop it. This is a bit quick & dirty but we don't want to
+      // get into calling more sophisticated functions in a cache clear and the table names are pretty unique.
+      // A separate process will reap the UserJobs but here the goal is just not to delete them during cache clearing
+      // if they are still referenced.
+      if (!CRM_Core_DAO::executeQuery("SELECT count(*) FROM civicrm_user_job WHERE metadata LIKE '%" . $tableDAO->tableName . "%'")) {
+        // drop leftover temporary tables
+        CRM_Core_DAO::executeQuery("DROP TABLE $table");
+      }
```

Comment:
FWIW, I think this variation would be relatively safe within a cache-clear:

```php
$liveTables = array_map(
  function($row) {
    // Note: Runs during cache-clear, so keep the lookups simple.
    $metadata = json_decode($row['metadata'] ?? '[]', 1);
    return $metadata['DataSource']['table_name'] ?? NULL;
  },
  CRM_Core_DAO::executeQuery("SELECT metadata FROM civicrm_user_job")->fetchAll()
);

$tables = array_diff($tables, $liveTables);
```

Reply:
Oh! It's MySQL 5.7! FWIW, one doesn't need `array_map` or `json_decode`:

```php
$liveTables = CRM_Core_DAO::executeQuery('SELECT JSON_EXTRACT(metadata, "$.DataSource.table_name") as tbl from civicrm_user_job')
  ->fetchMap('tbl', 'tbl');
```

Reply:
I'm kinda on the fence as to whether it is better - the import_table name is pretty distinct so it wouldn't be 'just mentioned' there & the array structure isn't so amazingly set in stone it could never change